cpp/tensorrt_llm/kernels/trtllmGenKernels/blockScaleMoe/RoutingKernel.h

```diff
@@ -26,14 +26,23 @@
 namespace moe::dev
 {
 
-////////////////////////////////////////////////////////////////////////////////////////////////////
-
 namespace routing
 {
-////////////////////////////////////////////////////////////////////////////////////////////////////
 
 namespace tg = batchedGemm::trtllm::gen;
 
+template <typename TypeExpW>
+struct PackedScoreIdx
+{
+    TypeExpW score;
+    int16_t idx; // @TODO: Might use int8_t as the number of experts is 128
```

Comment:
note: I thought about this before, but because there are only a few possibilities for the score:

- 4-byte floating point (or more like double): it doesn't matter whether we use 1-byte or 2-byte for idx because `PackedScoreIdx` cannot fit in register / coupled instructions anyway
- 2-byte floating point (or lower): it doesn't matter whether we use 1-byte or 2-byte for idx because it will always fit in 4 bytes, and there are no better instructions for e.g. 2-byte structures compared to 4-byte